### PR TITLE
3978 Add new Flume-Avro logger driver

### PR DESCRIPTION
**Pull request by sebglon**
**\- What I did**
Adding ne logger plugin for Flume Avro protocol
**\- How I did it**
#23978

Send log container message on Apache flume with [Avro protocol](https://avro.apache.org/docs/1.7.4/spec.html#binary_encoding)
add new library sebglon/goavro and snappy library
**\- How to verify it**
Run docker with new flume-avro logger
`dockerd  --log-driver=flume-avro --log-opt avro-host=<IP of flume> --log-opt avro-port=<flume port>`
Run container
`docker run --rm hello-world`
And show log message on your flume

Example AvroFlumeEvent: 

```
{ headers:{image_name=hello-world, container_name=/boring_heyrovsky, container_created=1472212849, daemon_name=docker, host=439c0ae46a3d, source=stdout, image_id=sha256:c54a2cc56cbb2f04003c1cd4507e118af7c0d340fe7e2720f70976c4b75237dc, partial=false, command=/hello, container_id=40682c62d234bf3b2218e5536973253afdbc68f06bcc5428e819a36e3bf8d015, timestamp=2016-08-26 12:00:49.935212928 +0000 UTC} body: 54 68 69 73 20 6D 65 73 73 61 67 65 20 73 68 6F This message sho }
```

**\- Description for the changelog**

New logger driver for [Apache Flume](https://flume.apache.org/)

### CODE REVIEW

**Comment by sebglon** [comment]
I work on docs

**Comment by sebglon** [comment]
Doc is OK for testing

**Comment by sebglon** [comment]
Can i have help with `windowsRS1` Failed?
This test work well on my laptop.

**Comment by sebglon** [comment]
All check is OK

**Comment by sebglon** [comment]
@vdemeester  Is ti possible to have a review. this is important for my client to user docker on prod.
The docs explain how this can be tested

File: api/client/plugin/inspect.go [hunk 1]
```diff
@@ -3,50 +3,57 @@
 package plugin
 
 import (
-	"encoding/json"
 	"fmt"
 
 	"github.com/docker/docker/api/client"
+	"github.com/docker/docker/api/client/inspect"
```
Comment (vdemeester) [inline api/client/plugin/inspect.go]: Hum these changes seems out of place :sweat_smile:

File: docs/reference/commandline/plugin_inspect.md [hunk 2]
```diff
@@ -17,7 +17,8 @@ Usage:  docker plugin inspect PLUGIN
 Inspect a plugin
 
 Options:
-      --help   Print usage
+      -f, --format string   Format the output using the given go template
```
Comment (vdemeester) [inline docs/reference/commandline/plugin_inspect.md]: So is this one…

File: integration-cli/docker_cli_plugins_test.go [hunk 3]
```diff
@@ -28,17 +26,7 @@ func (s *DockerSuite) TestPluginBasicOps(c *check.C) {
...
-	tmpFile, err := ioutil.TempFile("", "inspect.json")
-	c.Assert(err, checker.IsNil)
-	defer tmpFile.Close()
-
-	if _, err := tmpFile.Write([]byte(out)); err != nil {
-		c.Fatal(err)
-	}
-	// FIXME: When `docker plugin inspect` takes a format as input, jq can be replaced.
-	id, err := exec.Command("jq", ".Id", "--raw-output", tmpFile.Name()).CombinedOutput()
+	id, _, err := dockerCmdWithError("plugin", "inspect", "-f", "{{.Id}}", pNameWithTag)
```
Comment (vdemeester) [inline integration-cli/docker_cli_plugins_test.go]: … and this one

**Comment by vdemeester** [comment]
Design seems good to me. It would be easier for all parties to have a logging plugin infrastructure than having to add it in directly in docker for each (would make the maintenance easier).

/cc @cpuguy83 @LK4D4

**Pull request closed without merging**